### DISCUSSION TITLE
Why vim search exclude the letter in current cursor?

### DISCUSSION THREAD

**Discussion by ElwinGao-Microsoft** [vim/vim#14871] [Q&A] [answered]
when I search text like this:
--------------------------------
1. www.github.com
2. www.google.com
--------------------------------
current cursor(cursor type is block) on the first "w" in first line，and then I using "/" to search target words: "www"
when I type "enter" to search，the cursor jump to the first "w" in second line
I think vim will scan word from current letter(not jump to next "www"), but vim will search word exclude current letter
I don't understand why design like this?

**Comment by vim-ml** [comment, marked as answer]
On 2024-05-29, ElwinGao4444 wrote:
> when I search text like this:
> 
>  1. www.github.com
>  2. www.google.com
> 
> -------------------------------------------------------------------------------
> 
> current cursor(cursor type is block) on the first "w" in first line，and then I
> using "/" to search target words: "www"
> when I type "enter" to search，the cursor jump to the first "w" in second line
> I think vim will scan word from current letter(not jump to next "www"), but vim
> will search word exclude current letter
> I don't understand why design like this?

If Vim included the current location in the search and stopped at
the first "w" of the "www" in the first line, typing "n" to find the
next match would find that same match at the same location and the
search would never progress.

Regards,
Gary
